lbz/request.py

```diff
@@ -84,10 +103,48 @@ def json_body(self) -> dict | None:
                 raise BadRequestError(f"Content-Type header is missing or wrong: {content_type}")
         return self._json_body
 
-    def to_dict(self) -> dict:
-        copied = {k: v for k, v in self.__dict__.items() if not k.startswith("_")}
-        copied["headers"] = dict(copied["headers"])
-        copied["user"] = repr(copied["user"])
-        if copied["query_params"] is not None:
-            copied["query_params"] = dict(copied["query_params"])
-        return copied
+
+class WebSocketRequest(BaseRequest):
+    """Represents request from Web Socket Secure API Gateway."""
+
+    def __init__(
+        # pylint: disable=too-many-positional-arguments
+        self,
+        body: str | bytes | dict,
+        request_details: dict,
+        context: dict,
+        is_base64_encoded: bool,
+        user: User | None = None,
+        headers: CIMultiDict | None = None,
+    ):
+        self.headers = headers
+        self.context = context
+        self.user = user
+        self.action = request_details.pop("routeKey")
+        self.action_type = request_details.pop("eventType")
+        self.connection_id = request_details.pop("connectionId")
+        self.domain = request_details.pop("domainName")
+        self.stage = request_details.pop("stage")
+        self.details = request_details
+        super().__init__(body=body, is_base64_encoded=is_base64_encoded)
+
+    def __repr__(self) -> str:
+        return f"<Request {self.action_type} - {self.action} >"
+
+    def is_connection_request(self) -> bool:
+        return self.action_type is ActionType.CONNECT
+
+    def is_disconnection_request(self) -> bool:
+        return self.action_type is ActionType.DISCONNECT
+
+    @property
+    def json_body(self) -> dict | None:
```

Comment:
I started thinking whether these two classes are really that similar to inherit from the same base one 🤔

Reply:
The remaining methods in BaseRequest makes sense to have shared ancestor. It might still change before we actually release 0.7.0

Reply:
Ok, we will return to this comment in the next pull requests 👍

Reply:
FYI: #105